src/rdkafka_admin.h

```diff
@@ -0,0 +1,257 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _RDKAFKA_ADMIN_H_
+#define _RDKAFKA_ADMIN_H_
+
+
+#include "rdstring.h"
+#include "rdkafka_confval.h"
+
+
+
+/**
+ * @brief Common AdminOptions type used for all admin APIs.
+ *
+ * @remark Visit AdminOptions_use() when you change this struct
+ *         to make sure it is copied properly.
+ */
+struct rd_kafka_AdminOptions_s {
+        /* Generic */
+        rd_kafka_confval_t request_timeout;/**< I32: Full request timeout,
+                                            *        includes looking up leader
+                                            *        broker,
+                                            *        waiting for req/response,
+                                            *        etc. */
+        rd_ts_t abs_timeout;               /**< Absolute timeout calculated
+                                            *   from .timeout */
+
+        /* Specific for one or more APIs */
+        rd_kafka_confval_t operation_timeout; /**< I32: Timeout on broker.
+                                               *   Valid for:
+                                               *     CreateTopics
+                                               *     DeleteTopics
+                                               */
+        rd_kafka_confval_t validate_only;  /**< BOOL: Only validate (on broker),
+                                            *   but don't perform action.
+                                            *   Valid for:
+                                            *     CreateTopics
+                                            *     CreatePartitions
+                                            *     AlterConfigs
+                                            */
+
+        rd_kafka_confval_t incremental;    /**< BOOL: Incremental rather than
+                                            *         absolute application
+                                            *         of config.
+                                            *   Valid for:
+                                            *     AlterConfigs
+                                            */
+
+        rd_kafka_confval_t broker;         /**< INT: Explicitly override
+                                            *        broker id to send
+                                            *        requests to.
+                                            *   Valid for:
+                                            *     all
+                                            */
+
+        rd_kafka_confval_t opaque;         /**< PTR: Application opaque.
+                                            *   Valid for all. */
+};
+
+
+
+
+
+/**
+ * @name CreateTopics
+ * @{
+ */
+
+/**
+ * @brief NewTopic type, used with CreateTopics.
+ */
+struct rd_kafka_NewTopic_s {
```

Comment:
I'm going to call this `NewTopicSpecification` - which is what it actually is - in C# and hope no one notices i've deviated from the other languages.

Reply:
I strongly urge you to keep with the original Java API names unless there is a clear reason not to.
Inspiration:
https://docs.confluent.io/current/clients/javadocs/org/apache/kafka/clients/admin/AdminClient.html

Reply:
yeah, I've been following that.